### PR TITLE
MEX-402 Patient search fuzzy matcher should work for Spanish

### PR DESCRIPTION
**Pull request by brandones**
https://pihemr.atlassian.net/browse/MEX-402

### CODE REVIEW

**Comment by mogoodrich** [comment]
Seems cool and simple to do if we get it to work! :)

File: api/src/main/resources/hibernate.cfg.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+-->
+<!DOCTYPE hibernate-configuration PUBLIC
+        "-//Hibernate/Hibernate Configuration DTD 3.0//EN"
+        "http://www.hibernate.org/dtd/hibernate-configuration-3.0.dtd">
+
+<hibernate-configuration>
+
+    <session-factory>
+        <property name="hibernate.search.model_mapping">org.openmrs.module.pihcore.search.PihLuceneAnalyzerFactory</property>
+    </session-factory>
```
Comment (mseaton) [inline api/src/main/resources/hibernate.cfg.xml]: I don't know that a module can override the hibernate.cfg.xml file like this.  Do you know that this is possible?

File: api/src/main/java/org/openmrs/module/pihcore/search/PihLuceneAnalyzerFactory.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.apache.lucene.analysis.standard.ClassicFilterFactory;
+import org.hibernate.search.annotations.Factory;
+import org.hibernate.search.cfg.SearchMapping;
+import org.openmrs.api.db.hibernate.search.LuceneAnalyzers;
+
+
+public class PihLuceneAnalyzerFactory {
+    @Factory
+    public SearchMapping getSearchMapping() {
+        SearchMapping mapping = new SearchMapping();
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/pihcore/search/PihLuceneAnalyzerFactory.java]: One approach would be for this SearchMapping object to be defined as a Spring bean, where there is a default bean supplied by core, and modules can override this by adding their own bean that implements the same interface and the code within here chooses one vs. the other based on some configured sort order.   There are examples of doing this within the reporting module and elsewhere that I could point you to.  Not sure if that is the most efficient model or not, but it's an option.

Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/pihcore/search/PihLuceneAnalyzerFactory.java]: @mseaton @brandones this seems like a good approach to me!

**Pull request closed without merging**